Lib/test/test_embed.py

```diff
@@ -666,5 +666,16 @@ def test_preinit_isolated2(self):
         self.check_config("preinit_isolated2", config, preconfig)
 
 
+class AuditingTests(EmbeddingTestsMixin, unittest.TestCase):
```

Comment:
guard with ``hasattr(sys, 'addaudithook')``?

Reply:
We've failed to build test_embed if the hooks aren't available, so the whole test is broken. This file probably needs an "if not CPython: skip" condition.

Reply:
Which I'm not going to add right now, since we've clearly decided not to have one thus far :)